scripts/quests/otherAreas/RQ1_Rycharde_the_Chef.lua

```diff
@@ -0,0 +1,167 @@
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] = quest:progressEvent(70, xi.items.DHALMEL_MEAT),
+            ['Take'] = quest:event(68), -- Optional dialog.
+
+            onEventFinish =
+            {
+                [70] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    if option == 71 or option == 72 then -- Accept quest.
+                        quest:begin(player)
+                    elseif option == 70 then -- Decline quest.
+                        quest:setVar(player, 'Prog', 3)      
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: (OPTIONAL) Talk with Rycharde if you declined. Get quest (or decline again).
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and vars.Prog == 3
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] = quest:progressEvent(71, xi.items.DHALMEL_MEAT), -- You rejected the quest and you ask again.
+
+            onEventFinish =
+            {
+                [71] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    if option == 71 or option == 72 then  -- Accept quest.
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted. Handle trade outcomes. Handle talk with Rycharde.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:event(72) -- Interaction dialog.
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, {{xi.items.DHALMEL_MEAT, 2}}) then
+                        return quest:progressEvent(74) -- Quest completed dialog.
+                    elseif npcUtil.tradeHasExactly(trade, {{xi.items.DHALMEL_MEAT, 1}}) then
+                        return quest:event(73) -- "That's not enough!" dialog.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [74] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                        quest:setVar(player, 'DayCompleted', VanadielUniqueDay()) -- Used for next quest wait time.
+                    end
+                end,
+            },
+        },
+    },
+
+        -- Section: Quest completed. Change default message for Rycharde.
+    {
+        check = function(player, status, vars)
```

Comment:
Won't this check possibly override the check for the next quest to start since this only checks if completed this quest and next is avil it would create a race competition if it hits this file or the next. My opinion would be to remove this section and just add it to the next quests file with if wait time isn't achieved this cs else begin quest cs